apps/prairielearn/src/pages/instructorCourseAdminStaff/instructorCourseAdminStaff.html.ts

```diff
@@ -0,0 +1,757 @@
+import { z } from 'zod';
+
+import { escapeHtml, html } from '@prairielearn/html';
+import { renderEjs } from '@prairielearn/html-ejs';
+
+import { compiledScriptTag } from '../../lib/assets.js';
+import {
+  CourseInstance,
+  CourseInstanceSchema,
+  CourseInstancePermissionSchema,
+  CoursePermissionSchema,
+  User,
+  UserSchema,
+} from '../../lib/db-types.js';
+
+export const CourseUsersRowSchema = z.object({
+  user: UserSchema,
+  course_permission: CoursePermissionSchema,
+  course_instance_roles: z
+    .array(
+      z.object({
+        id: CourseInstanceSchema.shape.id,
+        short_name: CourseInstanceSchema.shape.short_name,
+        course_instance_permission_id: CourseInstancePermissionSchema.shape.id,
+        course_instance_role: CourseInstancePermissionSchema.shape.course_instance_role,
+        course_instance_role_formatted: z.string(),
+      }),
+    )
+    .nullable(),
+  other_course_instances: z
+    .array(
+      z.object({
+        id: CourseInstanceSchema.shape.id,
+        short_name: CourseInstanceSchema.shape.short_name,
+      }),
+    )
+    .nullable(),
+});
+type CourseUsersRow = z.infer<typeof CourseUsersRowSchema>;
+
+function hasUnknownUsers(courseUsers: CourseUsersRow[]) {
+  let hasUnknownUsers = false;
+  courseUsers.forEach((courseUser) => {
+    if (!courseUser.user.name) {
+      hasUnknownUsers = true;
+    }
+  });
+  return hasUnknownUsers;
```

Comment:
This can be written more compactly (and technically more efficiently) via [`Array.prototype.some`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some):

```suggestion
  return courseUsers.some(user => courseUser.user.name == null);
```

Reply:
Sorry, I messed up the variable names a bit, should be `courseUser` as the name of the function parameter.

Reply:
Updated to this in [100f8f0](https://github.com/PrairieLearn/PrairieLearn/pull/9822/commits/100f8f0527f2cffb292e278185a71aee2c615c06) and changed to `courseUser` in [dedfd29](https://github.com/PrairieLearn/PrairieLearn/pull/9822/commits/dedfd2957a59e4874fa9d38b80c389dbcba67da9).